src/java.base/share/classes/java/util/concurrent/ConcurrentHashMap.java

```diff
@@ -1020,7 +1021,8 @@ public V put(K key, V value) {
 
     /** Implementation for put and putIfAbsent */
     final V putVal(K key, V value, boolean onlyIfAbsent) {
-        if (key == null || value == null) throw new NullPointerException();
+        requireNonNull(key, "Provided key cannot be null!");
+        requireNonNull(value, "Provided value cannot be null!");
```

Comment:
Cool！

Reply:
Thanks. I was chasing my tail for hours because of this, so I wanted to add a message, to make things less ambiguous. It makes code safer and easier, imo.